src/clm/functions.py

```diff
@@ -137,19 +137,28 @@ def get_rdkit_fingerprints(mols, include_none=False):
     return fps
 
 
-def read_file(smiles_file, max_lines=None):
-    lines = [line for line in read_file_incremental(smiles_file, max_lines)]
+def read_file(smiles_file, max_lines=None, smile_only=False):
+    lines = [line for line in read_file_incremental(smiles_file, max_lines, smile_only)]
     lines_array = np.array(lines)
     return lines_array
 
 
-def read_file_incremental(input_file, max_lines=None):
+def read_file_incremental(
+    input_file,
+    max_lines=None,
+    smile_only=False,
+):
     count = 0
-    for line in open(input_file, "r").readlines():
-        yield line.strip()
-        count += 1
-        if max_lines is not None and count == max_lines:
-            return
+    with open(input_file, "r") as f:
+        # Extract index of smiles from header
+        if smile_only:
```

Comment:
Again, `smile_only` is an indication by the caller on what they want. It doesn't tell you whether the file contains smiles or it is a csv file.